src/arcade/potts/util/Plane.java

```diff
@@ -0,0 +1,78 @@
+package arcade.potts.util;
+
+import sim.util.Int3D;
+import arcade.potts.env.location.Voxel;
+
+/**
+ * A plane in 3D space.
+ */
+
+public final class Plane {
+    /** A point on the plane. */
+    public final Voxel referencePoint;
+    
+    /** The normal vector to the plane. */
+    public final Int3D normalVector;
+    
+    /**
+     * Creates a plane from a point and a vector.
+     *
+     * @param voxel  a point on the plane
+     * @param normalVector  the normal vector to the plane
+     */
+    public Plane(Voxel voxel, Int3D normalVector) {
+        this.referencePoint = voxel;
+        this.normalVector = normalVector;
+    }
+    
+    /**
+     * Determines distance from a point to the plane.
+     *
+     * @param p  the point
+     * @return  the distance from the point to the plane.
+     *          The distance is positive if the point is on
+     *          the same side of the plane as the normal vector
+     *         and negative if it is on the opposite side.
+     */
+    public double signedDistanceToPlane(Voxel p) {
+        double dotProduct;
+        double normalVectorMagnitude;
+        
+        dotProduct = (p.x - referencePoint.x) * normalVector.getX()
+                + (p.y - referencePoint.y) * normalVector.getY()
+                + (p.z - referencePoint.z) * normalVector.getZ();
+        normalVectorMagnitude = Math.sqrt(normalVector.getX() * normalVector.getX()
```

Comment:
Consider having the normal vector magnitude calculation as a separate method and/or normalize the normal vector in the constructor so you don't need to re-run the calculation for every voxel distance.